src/main/kotlin/com/atlassian/performance/tools/jiraactions/api/measure/ActionMeter.kt

```diff
@@ -140,67 +129,43 @@ class ActionMeter private constructor(
                 result = actionResult
             ).virtualUser(virtualUser)
             postMetricHook.run(actionMetricBuilder)
-            conditionallyAddDrilldown(actionMetricBuilder)
             output.write(actionMetricBuilder.build())
             throw Exception("Action '${key.label}' $actionResult", e)
         }
     }
 
-    private fun conditionallyAddDrilldown(actionMetricBuilder: ActionMetric.Builder): ActionMetric {
-        return actionMetricBuilder.build().let {
-            if (drilldownCondition.test(it)) {
-                actionMetricBuilder
-                    .drilldown(w3cPerformanceTimeline.record())
-                    .build()
-            } else {
-                it
-            }
-        }
-    }
-
     @Deprecated(message = "Use ActionMeter.Builder instead")
     fun withW3cPerformanceTimeline(
         w3cPerformanceTimeline: W3cPerformanceTimeline
     ): ActionMeter = ActionMeter(
         virtualUser = virtualUser,
         output = output,
         clock = clock,
-        w3cPerformanceTimeline = w3cPerformanceTimeline,
-        drilldownCondition = Predicate { true },
         postMetricHook = NoopPostMetricHook()
     )
 
     class Builder(
         private val output: ActionMetricOutput
     ) {
         private var virtualUser: UUID = UUID.randomUUID()
-        private var w3cPerformanceTimeline: W3cPerformanceTimeline = DisabledW3cPerformanceTimeline()
-        private var drilldownCondition: Predicate<ActionMetric> = Predicate { true }
         private var clock = Clock.systemUTC()
         private var postMetricHook: PostMetricHook = NoopPostMetricHook()
 
         constructor(meter: ActionMeter) : this(
             meter.output
         ) {
             clock = meter.clock
-            w3cPerformanceTimeline = meter.w3cPerformanceTimeline
-            drilldownCondition = meter.drilldownCondition
             virtualUser = meter.virtualUser
         }
 
-        fun performanceTimeline(w3cPerformanceTimeline: W3cPerformanceTimeline) = apply { this.w3cPerformanceTimeline = w3cPerformanceTimeline }
-        fun drilldownCondition(drilldownCondition: Predicate<ActionMetric>) = apply { this.drilldownCondition = drilldownCondition }
         fun clock(clock: Clock) = apply { this.clock = clock }
         fun postMetricHook(postMetricHook: PostMetricHook) = apply { this.postMetricHook = postMetricHook }
```

Comment:
It's easy to overwrite the single hook. It's hard to add multiple ones (without discarding the old ones).

- [ ] consider `appendPostMetricHook` in the builder